src/java/voldemort/utils/Props.java

```diff
@@ -128,7 +128,11 @@ public String put(String key, String value) {
     }
 
     public String put(String key, Integer value) {
-        return props.put(key, value.toString());
+        if (null == value) {
```

Comment:
This is inline with the already existing `getNullableInt()` function which returns null if the key is not contained.

The previous code did not allow passing in a null value at all, since it results in a NPE at the time the `put` is called. As far as I can tell, there is no code that depends on receiving a NPE when passing in a null.

That being said, I agree that this isn't super clean. Perhaps I should create a new config rather than overloading the meaning of the existing config.